crates/router/src/connector/worldpay/transformers.rs

```diff
@@ -271,11 +294,24 @@ impl TryFrom<types::PaymentsResponseRouterData<WorldpayPaymentsResponse>>
     }
 }
 
-impl<F> TryFrom<&types::RefundsRouterData<F>> for WorldpayRefundRequest {
+impl TryFrom<&types::PaymentsCaptureRouterData> for WorldpayPartialRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            reference: item.payment_id.clone().replace("_", "-"),
+            value: PaymentValue {
+                amount: item.request.amount_to_capture,
```

Comment:
amount should be consumed for connector routerdata. Please convert the amount before passing it to transformers.
```suggestion
                amount: item.amount_to_capture,
```